src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectMethodExecutor.cs

```diff
@@ -67,6 +73,39 @@ public object Execute(object target, object[] parameters)
             return _executor(target, parameters);
         }
 
+        public object GetDefaultValueForParameter(int index)
+        {
+            Debug.Assert(ActionParameters.Length > 0);
+
+            var parameterInfo = ActionParameters[index];
+            if (_parameterDefaultValues == null)
+            {
+                _parameterDefaultValues = new object[ActionParameters.Length];
+            }
+
+            var defaultValue = _parameterDefaultValues[index];
+            if (defaultValue == null)
```

Comment:
Need a separate guard for this logic since repeating the same logic to come up w/ `null` isn't worthwhile.